### PR TITLE
refactor/1764 - Fleet Retirements

### PR DESCRIPTION
**Pull request by Hawaiicoder**
The purpose of this pull request is to bring airline fleets up to date in light of recent retirements,

### CODE REVIEW

**Comment by Hawaiicoder** [comment]
If anyone has any insight on why the build is failing, would you be kind enough to advise me?

**Comment by cake-pie** [comment]
extra comma, klm line 45

**Comment by Hawaiicoder** [comment]
You're excellent at this!

File: assets/airlines/klm.json [hunk 1]
```diff
@@ -1,72 +1,69 @@
...
+            ["E190", 32]
+        ],
+        "short": [
+            ["B737", 18],
+            ["B738", 27],
+            ["B739", 5]
+        ],
+        "tncm": [
+            ["A332", 8],
+            ["A333", 5],
```
Comment (n8rzz) [inline assets/airlines/klm.json:45]: this is why the build isn't happy

Comment (Hawaiicoder) [reply assets/airlines/klm.json]: Fixed.

File: assets/airlines/aal.json [hunk 2]
```diff
@@ -1,67 +1,56 @@
...
-            ["A320", 48],
-            ["A321", 219],
-            ["B38M", 24],
-            ["B738", 304],
-            ["E190", 19]
-        ],
-        "tncmShort": [
-            ["B738", 304]
-        ],
-        "tncmLong": [
```
Comment (indianbhaji) [inline assets/airlines/aal.json:52]: As you have removed this fleet, you need to remove any references to this fleet in any airport files, in this case, TNCM.

File: assets/airlines/aal.json [hunk 3]
```diff
@@ -1,67 +1,56 @@
...
-            ["B752", 34],
-            ["B763", 17]
-        ],
-        "a321": [
-            ["A21N", 10],
-            ["A321", 219]
-        ],
-        "a332": [
-            ["A332", 15]
-        ],
```
Comment (indianbhaji) [inline assets/airlines/aal.json:62]: This fleet is used in EDDM.

File: assets/airlines/afr.json [hunk 4]
```diff
@@ -67,9 +62,6 @@
         "a321": [
             ["A321", 20]
         ],
-        "a343": [
```
Comment (indianbhaji) [inline assets/airlines/afr.json:70]: Used in a couple of airports - check and remove references.

File: assets/airlines/baw.json [hunk 5]
```diff
@@ -1,75 +1,65 @@
...
-        ],
-        "short": [
-            ["A20N", 10],
-            ["A21N", 8],
-            ["A318", 1],
-            ["A319", 39],
-            ["A320", 67],
-            ["A321", 18]
-        ],
-        "a318": [
```
Comment (indianbhaji) [inline assets/airlines/baw.json:51]: Used in KJFK

File: assets/airlines/baw.json [hunk 6]
```diff
@@ -1,75 +1,65 @@
...
-            ["A318", 39]
-        ],
-        "a319": [
-            ["A319", 39]
-        ],
-        "a321": [
-            ["A21N", 8],
-            ["A321", 18]
-        ],
-        "b744": [
```
Comment (indianbhaji) [inline assets/airlines/baw.json:61]: Used in KAUS

**Review by indianbhaji — CHANGES_REQUESTED** [review]
When you remove fleets, make sure you check whether they are being used in any airport files as this will cause issues when they are run :)

File: assets/airlines/dal.json [hunk 7]
```diff
@@ -1,70 +1,61 @@
...
+            ["B739", 130]
+        ],
+        "b763": [
+            ["B763", 49]
+        ],
+        "b764": [
+            ["B764", 21]
+        ]
+    }
+}
```
Comment (cake-pie) [inline assets/airlines/dal.json:61]: missing newline at end of file

**Review by cake-pie — CHANGES_REQUESTED** [review]
Several files had line endings were changed from unix to windows. 

Some fleets were deleted entirely. You will need to ensure that they are not still in use by some airports, and update such airports accordingly -- the airline may have switched to using a different aircraft type there, or may have stopped serving the airport entirely.

**Comment by Hawaiicoder** [comment]
> Several files had line endings were changed from unix to windows.
> 
> Some fleets were deleted entirely. You will need to ensure that they are not still in use by some airports, and update such airports accordingly -- the airline may have switched to using a different aircraft type there, or may have stopped serving the airport entirely.

Should I update these based on the current situation? If so, then basically every airport has an unrealistic traffic level. I would assume the best route to take is to find information from _just_ prior to the pandemic. Do you agree?

**Comment by indianbhaji** [comment]
> > Several files had line endings were changed from unix to windows.
> > Some fleets were deleted entirely. You will need to ensure that they are not still in use by some airports, and update such airports accordingly -- the airline may have switched to using a different aircraft type there, or may have stopped serving the airport entirely.
> 
> Should I update these based on the current situation? If so, then basically every airport has an unrealistic traffic level. I would assume the best route to take is to find information from _just_ prior to the pandemic. Do you agree?

It is enough to just simply find a similar fleet for that airline or remove the fleet specification (ie. /b752) completely. We don't want to be messing with traffic levels in this pull request.

File: assets/airlines/aal.json (hunk 2 above)
Comment (Hawaiicoder) [reply assets/airlines/aal.json:52]: Looked through TNCM, there's no reference to the deleted fleet.

File: assets/airlines/aal.json (hunk 3 above)
Comment (Hawaiicoder) [reply assets/airlines/aal.json:62]: Could not find any specific references to deleted fleets in EDDM.

**Comment by cake-pie** [comment]
> Thought I did this before but I suppose that's not the case...

You did, and then removed it again in the very next commit.

**Review by cake-pie — CHANGES_REQUESTED** [review]
> line endings were changed from unix to windows.

still not fixed.

File: assets/airlines/aal.json (hunk 3 above)
Comment (Marcel510) [reply assets/airlines/aal.json:62]: See eddm.json line 2373 for example

**Comment by Hawaiicoder** [comment]
I'm going to close this- I don't really have time to work on anything for Openscope at the moment, and what little time I do get, I'd like to spend on BWI. Thanks to everyone for the help.

**Pull request closed without merging**